src/sqlfluff/dialects/dialect_redshift.py

```diff
@@ -2401,3 +2401,11 @@ class FunctionSegment(ansi.FunctionSegment):
             ),
         ),
     )
+
+
+class ColumnReferenceSegment(postgres.ColumnReferenceSegment):
+    """A reference to column, field or alias.
+
+    We override this for Redshift to allow keywords in fully qualified column
+    names (using Full segments) like it's done in Postgres.
+    """
```

Comment:
I think you're right that this should just refer to postgres - but not by inserting it here.

The redshift dialect already should inherit everything direct from postgres. The issue here is that it seems that someone is already overriding that on line 180 of `dialect_redshift.py`. I think if you delete those lines you should be able to get the same effect without needing to effectively double-override it here.

Reply:
Good call! I didn't spot line 180. I'll apply the changes and run the tests :+1: